### PR TITLE
DACCESS-352 - Don't generate html for advanced search form in helpers

### PR DESCRIPTION
**Pull request by chrisrlc**
https://culibrary.atlassian.net/browse/DACCESS-352

- Rewrites Advanced Search form views (index and edit) so that form html is in views, instead of dynamically generated via helper methods in AdvancedHelper.
   - `#index` might not be the best fit for the action, but left it the same to match search
- Uses more of rails Action View form helpers in views, instead of raw html
- Cleans up associated methods except for `render_edit_advanced_constraints_filters` in CornellParamsHelper, which gets into some code that really could use a review, but just didn't have time to get to it.
- Cleans up AdvancedHelper methods and writes unit tests
- Initial work for turning boolean radio buttons into select button, updated coffeescript that operates "Add a row" in form, updated corresponding cucumber tests
    - Coffeescript looked good, but could be a future project to convert the remaining few coffeescript files to modern js?
    - Needs further styling feedback from Melissa

Note: use of single quotes. I saw some PRs float by in recent weeks that change existing single quotes to double. Was there a team decision to use double quotes while I was out? I can switch them to double if so! If no, may bring up during retro. Either is fine, as long as we're consistent. Fwiw, [ruby style guide](https://github.com/rubocop/ruby-style-guide?tab=readme-ov-file#consistent-string-literals) uses single (but also says that only consistency matters). I personally like single because that's what everywhere else I've worked at has used, but I have no other good reason than that, so I don't matter!

### CODE REVIEW

**Review by bmckinney — APPROVED** [review]
wow, among other things, i love that you yanked out the html generating code from the helper

i learn a lot looking at your tests too, the rspec in particular!

i don't know much about the single vs double quote convention but agree about picking one and trying to stick to it. who made the commits you mention? in any case it reminds me of the Silicone Valley episode about indenting spaces vs tabs

File: blacklight-cornell/app/views/advanced_search/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<% all_form_values = params['action'] == 'edit' ? params_to_form_values(params) : default_form_values(params) %>
+<% all_hidden_form_values = params['action'] == 'edit' ? params_to_hidden_form_values(params) : [] %>
```
Comment (chrisrlc) [inline blacklight-cornell/app/views/advanced_search/_form.html.erb:2]: Huh, why did I do it this way. I could have just passed in the relevant values as locals when rendering the partials in index vs edit. I'll change this tomorrow.

**Review by sarah-cul — APPROVED** [review]
I agree with Bill, the advanced_helper file is so much cleaner and the code looks more organized!

**Review by mhk33 — APPROVED** [review]
Ahhh this looks so much better!

Agree with you about single or double quotes consistency, and single sounds good to me.

**Pull request merged**